src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -581,13 +579,54 @@ private void checkDataLength(int ... lengths) {
         }
     }
 
+    /**
+     * Intrinsic for Vector AES Galois Counter Mode implementation.
+     * AES and GHASH operations are interleaved in the intrinsic implementation.
+     * return - number of processed bytes
+     *
+     * Requires 768 bytes (48 AES blocks) to efficiently use the intrinsic
+     * @param in input buffer
+     * @param inOfs input offset
+     * @param ct buffer that ghash will read (in for encrypt, out for decrypt)
+     * @param ctOfs offset for ct buffer
+     * @param out output buffer
+     * @param outOfs output offset
+     * @param gctr object for the CTR operation
+     * @param ghash object for the ghash operation
+     */
+    @IntrinsicCandidate
+    private static int implGCMCrypt(byte[] in, int inOfs, int inLen,
+        byte[] ct, int ctOfs, byte[] out, int outOfs,
+        GCTR gctr, GHASH ghash) {
+
+        int len = 0;
+        if (inLen > TRIGGERLEN) {
+            int i = 0;
+            int segments = (inLen / 6);
+            segments -= segments % gctr.blockSize;
+            do {
+                len += gctr.update(in, inOfs + len, segments, out,
+                        outOfs + len);
+                ghash.update(ct, ctOfs, segments);
+                ctOfs = len;
```

Comment:
This does not look right when the initial value of ctOfs != 0.

Reply:
Yeah that doesn't look right